### ISSUE TITLE
Hot keys for play/pause/step

### ISSUE THREAD

**Issue by jbphet** [phetsims/scenery-phet#571]
As reported in https://github.com/phetsims/molecules-and-light/issues/262, some users of keyboard navigation and other a11y features have had some difficulties using the play/pause/step functionality.  Hot keys may be useful for this.  We have used some element-specific hot keys in other contexts, such as for moving balloons in BASE and moving the ruler in GFL, but we haven't done any global ones yet, so this may prove to be a challenge.

The initial investigation should focus on whether this would indeed be useful and, if so, what keys could be used that wouldn't interfere with the hot keys used by screen readers.  @terracoda and @Matthew-phet have volunteered to look into this.  If and when they have a solid recommendation, the issue should be assigned back to me and I'll work with the development group to prioritize.

**Comment by Matthew-Moore240** [comment]
YouTube: k | Pause/Play in player.

Twitch: It says space bar is the hotkey but I tested K and it works so I assume that its an HTML 5 thing.

**Comment by jessegreenberg** [comment]
Assigning to myself to explore feasibility of this. In general, global hot keys are not compatible with screen reading software, but we would like to revisit this. It is possible that using a modifier key or something like that would make this more possible.

**Comment by terracoda** [comment]
I found this Microsoft Stream resource on [Hot keys & Accessibility](https://docs.microsoft.com/en-us/stream/hotkeys-and-accessibility). It seems focus is important and they hot keys are not expected to work with screen reader software enabled. Screen reader users have their own customizable settings.

**Comment by terracoda** [comment]
Here are some resources I have read in the past:
- [J, K or How to choose keyboard shortcuts for web applications](https://medium.com/@sashika/j-k-or-how-to-choose-keyboard-shortcuts-for-web-applications-a7c3b7b408ee)
- [Time to revisit accesskey](https://tink.uk/time-to-revisit-accesskey)

Here are some resources that were shared by David Best on the w3-wai-ig list that might be helpful:
- Computer Hope - [Computer keyboard shortcut keys](https://www.computerhope.com/shortcut.htm)
- [ShortcutWorld.com](http://shortcutworld.coml) - An open, wiki-style Reference Database for Keyboard Shortcuts.

And here are some comments from the list:
From Jonathan Avila
> It makes sense.  Per WCAG 2.1 if you do rely on a single key shortcut make sure it can be turned off or changed.  Whether you need to use a modifier or not with a screen reader may depend on whether your application forces focus/forms mode or not via role application or similar technique.    If the screen reader is in browse mode you would need a keystroke that was not swallowed up by the screen reader such as a modifier that was also not a browser keystroke.  So you’d likely want to test with several different assistive technology and browser combinations.  Either way providing some flexibility for the user will allow them to change it if there is a conflict.

From Phill Jenkins:
> If you're referring to just your simulation app, where you control the single page of the app, then you are free to create hotkeys that are effective anywhere on the page without requiring focus to easily control the multiple different inputs and play/pause, step forward/stop, etc. 
>
> I was involved with a middle school level photosynthesis simulation app years ago that had many separate controls for quickly changing the sunlight, water, carbon dioxide, etc. as the simulation progressed.

**Comment by terracoda** [comment]
Avoid the modifier keys for JAWS, NVDA, and VO.

**Let's try:**
- Alt + k (k is often used for web-based video players)

@jessegreenberg to test in a basic html page first.

**Comment by jessegreenberg** [comment]
Here is some HTML to test:

<details>

```html
<!DOCTYPE html>
<html>
<head>
  <meta charset="UTF-8">
  <title>title</title>
</head>
<body>

<div id="parent">
  <h1>Some Heading</h1>
  <p>Some content to read</p>
  <button id="focusable">Focusable</button>

  <label for="slider">My Slider</label>
  <input type="range" id="slider">
</div>


</body>

<script>
  const button = document.getElementById( 'focusable' );
  const parent = document.getElementById( 'parent' );

  // add custom input listeners

  document.body.addEventListener( 'keydown', event => {
    console.log( 'event received' );
  } );
</script>
</html>
```
</details>

**Comment by jessegreenberg** [comment]
Testing results:
NVDA + Firefox - "Alt" + anything should work well.

**Comment by terracoda** [comment]
@jessegreenberg, yes, that sounds wonderful!

**Comment by Matthew-Moore240** [comment]
awesome! Cant wait to see it in action!

**Comment by jessegreenberg** [comment]
This was done in the above commits, but I would like to review the solution as part of https://github.com/phetsims/scenery/issues/1048 at the next a11y dev meeting with @zepumph.

**Comment by terracoda** [comment]
This works great @jessegreenberg. I tried several scenarios, and the only issue was the VoiceOver sometimes stops announcing responses, which is a VoiceOver issue not a hotkey issue.

I really like the flexibility of using the hotkey. I tried it from several different places and it seemed to work fine.

One awesome situation is when I paused the sim before using the Reset Molecule button, then my focus was automatically moved to the Light Source switch. I was able to quickly play the sim (un-pause it) with the Hotkey instead of having to navigate off the Light Source button all the way over to the Play button!!!

It's awesome!

**Comment by terracoda** [comment]
Over to you @Matthew-Moore240 for your enjoyment!

**Comment by Matthew-Moore240** [comment]
I agree its awesome! One thing I forgot about was if the hotkey would unpause if you hit it again glad to see that works perfect as well!

**Comment by jessegreenberg** [comment]
OK, thanks! Removing assignments and we will discuss the code changes at an upcoming meeting.

**Comment by terracoda** [comment]
@Matthew-Moore240 re your comment

> One thing I forgot about was if the hotkey would unpause if you hit it again glad to see that works perfect as well!

Yep, it works like a toggle button.

**Comment by terracoda** [comment]
@jessegreenberg, do mean a dev meeting or an a11y meeting?
I'll draft something for the Keyboard Shortcuts Dialog.

**Comment by jessegreenberg** [comment]
I meant a11y-dev meeting in https://github.com/phetsims/scenery-phet/issues/571#issuecomment-617862360, I want to review the code change for this issue and https://github.com/phetsims/scenery/issues/1048.

**Comment by jessegreenberg** [comment]
@zepumph and I reviewed this old issue during meeting today, in general the pattern looks OK for global listeners. There is some cleanup to do in phetsims/scenery#1048, but nothing else for this issue.
